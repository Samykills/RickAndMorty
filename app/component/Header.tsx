import React, { PureComponent } from 'react';
import { Animated, StyleSheet, Text, Platform } from 'react-native';

const HEADER_IMAGE = 'https://comicsalliance.com/files/2016/01/rick-and-morty.jpg';
const HEADER_MAX_HEIGHT = 200;
const HEADER_MIN_HEIGHT = Platform.OS == 'ios' ? 80 : 50;
const HEADER_SCROLL_DISTANCE = HEADER_MAX_HEIGHT - HEADER_MIN_HEIGHT;

export interface Props {
  scrollY: any;
}

class Header extends PureComponent<Props> {
  render() {
    const headerHeight = this.props.scrollY.interpolate({
      inputRange: [0, HEADER_SCROLL_DISTANCE],
      outputRange: [HEADER_MAX_HEIGHT, HEADER_MIN_HEIGHT],
      extrapolate: 'clamp'
    });
    const imageOpacity = this.props.scrollY.interpolate({
      inputRange: [0, HEADER_SCROLL_DISTANCE / 2, HEADER_SCROLL_DISTANCE],
      outputRange: [1, 1, 0],
      extrapolate: 'clamp'
    });
    const imageTranslate = this.props.scrollY.interpolate({
      inputRange: [0, HEADER_SCROLL_DISTANCE],
      outputRange: [0, -50],
      extrapolate: 'clamp'
    });
    const headerOpacity = this.props.scrollY.interpolate({
      inputRange: [0, HEADER_SCROLL_DISTANCE / 2, HEADER_SCROLL_DISTANCE],
      outputRange: [0, 0, 1],
      extrapolate: 'clamp'
    });

    return (
      <Animated.View style={[styles.header, { height: headerHeight }]}>
        <Animated.Image
          resizeMode={'cover'}
          style={[styles.backgroundImage, { opacity: imageOpacity, transform: [{ translateY: imageTranslate }] }]}
          source={{ uri: HEADER_IMAGE }}
        />
        <Animated.View style={[styles.bar, { opacity: headerOpacity }]}>
          <Text style={styles.title}>All Characters</Text>
        </Animated.View>
      </Animated.View>
    );
  }
}
const styles = StyleSheet.create({
  header: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    backgroundColor: '#212121',
    overflow: 'hidden'
  },
  bar: {
    marginTop: Platform.OS == 'ios' ? 38 : 15,
    height: 32,
    alignItems: 'center',
    justifyContent: 'center'
  },
  title: {
    backgroundColor: 'transparent',
    color: 'white',
    fontSize: 18
  },
  backgroundImage: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    width: undefined,
    height: HEADER_MAX_HEIGHT,
    resizeMode: 'cover'
  }
});
export default Header;
