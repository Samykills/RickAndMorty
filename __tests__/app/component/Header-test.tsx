// __tests__/Intro-test.js
import React from 'react';
import { Animated } from 'react-native';
import Header from '../../../app/component/Header';
import renderer from 'react-test-renderer';

test('Header renders correctly', () => {
  const tree = renderer.create(<Header scrollY={new Animated.Value(0)} />).toJSON();
  expect(tree).toMatchSnapshot();
});
