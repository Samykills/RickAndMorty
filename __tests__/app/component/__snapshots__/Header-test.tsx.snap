// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#212121",
      "height": 200,
      "left": 0,
      "overflow": "hidden",
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://comicsalliance.com/files/2016/01/rick-and-morty.jpg",
      }
    }
    style={
      Object {
        "height": 200,
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "resizeMode": "cover",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
        "width": undefined,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 32,
        "justifyContent": "center",
        "marginTop": 38,
        "opacity": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      All Characters
    </Text>
  </View>
</View>
`;
